src/auth/downscopedclient.ts

```diff
@@ -129,39 +172,42 @@ export class DownscopedClient extends AuthClient {
 
   /**
    * Provides a mechanism to inject Downscoped access tokens directly.
-   * When the provided credential expires, a new credential, using the
-   * external account options are retrieved.
-   * Notice DownscopedClient is the broker class mainly used for generate
-   * downscoped access tokens, it is unlikely we call this function in real
-   * use case.
-   * We implement to make this a helper function for testing all cases in getAccessToken().
+   * The expiry_date field is required to facilitate determination of the token
+   * expiration which would make it easier for the token consumer to handle.
    * @param credentials The Credentials object to set on the current client.
    */
   setCredentials(credentials: Credentials) {
+    if (!credentials.expiry_date) {
```

Comment:
I don't know about this. It is technically possible to get a downscoped access token without an expiry (source creds no expiry, STS does not return one). Would be weird to return a downscoped access token without expiry, but then they can't inject it here. Though, I'm not sure why a developer would even use this method.

@bojeil-google thoughts?

Reply:
`setCredentials` is defined in the abstract super class `AuthClient`, so it will always be there.
I doubt it will be used. Most customers will just provide a source `AuthClient`. In that case, we should have the ability to get the underlying expiration time from the source `AuthClient`. We are trying to make sure if the customer provides it to us (though unlikely), they also provide the expiration time to make things easier and more determinable. I guess we can always relax it in the future if a customer has a good reason for it, though I highly doubt this will be used.